### ISSUE TITLE
allow ignore on a per-argument basis for `reportIncompatibleMethodOverride`

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pyright#10440]
Consider the following code:
```python
# pyright: standard
class A:
    pass


class B(A):
    pass


class C:
    def func(self, a: A, i: int, s: str):
        pass


class D(C):
    def func(self, a: B, i2: int, s: str):
        pass
```

`pyright` reports:
```text
incompat.py:16:9 - error: Method "func" overrides class "C" in an incompatible manner
    Parameter 2 type mismatch: base parameter is type "A", override parameter is type "B"
    Parameter 3 name mismatch: base parameter is named "i", override parameter is named "i2"
      "A" is not assignable to "B" (reportIncompatibleMethodOverride)
```

This report does make sense to me, but I'd like to have more flexibility in how I can ignore the details of specific parameters.

**Describe the solution you’d like**

I would like to be able to ignore this message on a parameter by parameter basis.

If I use `# pyright: ignore[reportincompatibleoverride]`, I have to apply it to the entire `func()` method, but I'd like to be able to say "This parameter is OK for me for typing", i.e., ignore the mismatching that pyright is detecting selectively by parameter, as opposed to the whole function.  So maybe I could write:

```python
class D(C):
    def func(self, a: B,  
          i2: int,  # pyright: ignore[reportincompatibleoverride]
          s: str):
        pass
```
to indicate that I'm OK to ignore what happened with the `i2` argument, but not the others.

Note - this is not just about name mismatches versus type mismatches.  I may have type mismatches that I'm OK to ignore, not just name mismatches.

**Comment by erictraut** [comment]
I think it's unlikely that we'd implement this. There's no precedent for a "partial ignore", and I don't think it's a good idea to create such a precedent.

Ideally, you should avoid creating LSP violations, so fixing the code rather than ignoring cases like this should be the goal.
